0-module-and-module-source.emu

```diff
@@ -537,27 +569,48 @@ location: https://tc39.es/proposal-compartments/
         </ul>
 
         <emu-clause id="sec-module">
-            <h1>Module ( _moduleSource_, _referral_, _importHook_ [, _importMeta_ ] )</h1>
+            <h1>Module ( _moduleSource_, _handler_ [, _importMeta_ ] )</h1>
             <p>When the `Module` function is called with the arguments _moduleSource_, _referral_, _importHook_ and _importMeta_, the following steps are taken:</p>
             <emu-alg>
                 1. If NewTarget is *undefined*, throw a *TypeError* exception.
                 1. If Type(_moduleSource_) is not Object, throw a *TypeError* exception.
-                1. If IsCallable(_importHook_) is *false*, throw a *TypeError* exception.
-                1. If _importMeta_ is *undefined*, then
-                  1. Set _importMeta_ to *null*.
-                1. Else, if Type(_importMeta_) is not Object, then
-                  1. Throw a *TypeError* exception.
+                _importHook_
                 1. Perform ? RequireInternalSlot(_moduleSource_, [[ModuleSource]]).
-                1. Let _O_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%Module.prototype%"*, « [[Module]], [[ModuleSourceInstance]], [[ImportHook]] »).
                 1. Let _moduleRecord_ to ! CreateModuleRecord(_moduleSource_.[[ModuleSource]]).
-                1. Set _moduleRecord_.[[ModuleInstance]] to _O_.
-                1. Set _moduleRecord_.[[ImportMeta]] to _importMeta_.
-                1. Let _importHookClosure_ be a new Abstract Closure with parameters (_specifier_) that captures _importHook_ and _referral_ and performs the following steps when called
-                  1. Assert: Type(_specifier_) is String.
-                  1. return ? Call(_importHook_, *undefined*, « _specifier_, _referral_ »).
-                1. Set _O_.[[Module]] to _moduleRecord_.
+                1. If Type(_handler_) is Object, then
+                  1. If _handler_ is *null, throw a TypeError exception.
```

Comment:
```suggestion
                  1. If _handler_ is *null*, throw a TypeError exception.
```

Reply:
AFAIK, spec language `Type(_val_)` will not return `object` for `null`?

Reply:
Right, Type(null) is Null.

Reply:
yeah... I will fix it, it should be above this line.

Reply:
The only remaining question here is what to do if the handler is not null or object? should we throw? in the case of proxies, it is required... which makes things easier, but I don't know if we want to pay that cost here considering that if there are no hooks, the handler is useless.

Reply:
@nicolo-ribaudo can you elaborate more on this?